### PR TITLE
quantization: Linear + BatchNorm1d fusion

### PR DESCRIPTION
**Pull request by vkuzo**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#50748 quantization: Linear + BatchNorm1d fusion**

Summary:

Adds support for Linear + BatchNorm1d fusion to quantization.

This is a redo of @dreiss's https://github.com/pytorch/pytorch/pull/37467, faster
to copy-paste it than rebase and deal with conflicts.

Test Plan:

```
python test/test_quantization.py TestFusion.test_fusion_linear_bn_eval
```

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D25957432](https://our.internmc.facebook.com/intern/diff/D25957432)

### CODE REVIEW

File: torch/nn/utils/fusion.py [hunk 1]
```diff
@@ -26,3 +26,23 @@ def fuse_conv_bn_weights(conv_w, conv_b, bn_rm, bn_rv, bn_eps, bn_w, bn_b):
...
+
+    fused_linear.weight, fused_linear.bias = fuse_linear_bn_weights(
+        fused_linear.weight, fused_linear.bias,
+        bn.running_mean, bn.running_var, bn.eps, bn.weight, bn.bias)
+
+    return fused_linear
+
+def fuse_linear_bn_weights(linear_w, linear_b, bn_rm, bn_rv, bn_eps, bn_w, bn_b):
+    if linear_b is None:
+        linear_b = bn_rm.new_zeros(bn_rm.shape)
```
Comment (vadimkantorov) [inline torch/nn/utils/fusion.py:42]: should this be just `linear_b = torch.zeros_like(bn_rm)`?

**Review by supriyar — APPROVED** [review]

File: torch/nn/utils/fusion.py (hunk 1 above)
Comment (vkuzo) [reply torch/nn/utils/fusion.py]: thanks, fixed
